### ISSUE TITLE
Anybody continuous his own branch? I can join to develop.

### ISSUE THREAD

**Issue by fiftin** [angular-ui/bootstrap#6658]
Project development is stopped, as I understood.  May be anybody continuous his own branch? I can join to develop.

**Comment by navarroaxel** [comment]
@icfantv, can I be added as collaborator to continue with this repo ? I have some PRs opened from March.

**Comment by icfantv** [comment]
@navarroaxel I would strongly encourage you to have a frank discussion with some OSS project maintainers, preferably more than one and with very popular projects, to get a sense of what, exactly you're getting yourself into.  Managing a popular OSS repo is a full-time job in and of itself and it's primarily thankless.  It also usually takes multiple people.

You need to have high standards and you cannot let the desire to just get stuff done (or just please the community) interfere with maintaining those high standards.  Example, there are a TON of PRs which lack tests.

If you want to fork and maintain this repo, it's OSS, I cannot stop you.  If nothing else, it will be a huge, eye-opening experience for you.  But I'm not comfortable giving you commit privileges to this repo at this time.

I'll go ahead and leave this issue open for a little while to see if people see it and you can get folks on board.  I will say, however, that I don't feel continuing to maintain this is the right way to go.

**Comment by navarroaxel** [comment]
@icfantv I'm sure I'm not a hero and maybe I can't handle the size of this OSS. But the way this project was closed is sadly incorrect. :cry: 

The latest release was in January and no more news about Merges and releases until your latest update saying this was closed due to maintainers moved to other cooler or more rewarded projects. This project is used by a lot of people and there is still bugs in the demo page. I think critical and high priority issues should being fixed.
There area a lot of people who sent PRs since February fixing things and nothing was doing until your update saying this project is `feature-complete` with bugs in the own demo page.

I really don't know what to do, but if anyone forks and creates his own package we are dividing our efforts and disintegrating this awesome project. This is an awesome project because this replaced `jquery` :clap:. The close of this project is so sad to me, even when I'm moving to another non-deprecated technology.

Maybe we can build another group of maintainers with good reputation and this project can receive fixes in an enough good time of response. Any time is better than 10 months of nothing.

**Comment by icfantv** [comment]
@navarroaxel I was really debating how, or if, to respond to this.  Intentional or not, you touch on some of the enormous difficulties of running a popular OSS repo:

* There will always be people who disagree with you.  ALWAYS.  Responses will vary from taking it in stride to launching personal attacks because some people treat OSS like it's there to serve only their needs.  One needs to have a VERY thick skin on OSS projects.
* People [may] search/read, but they do not see.  People post issues asking for help even though the ISSUE_TEMPLATE indicates how to get help.  The PR template says exactly what we expect from PRs and yet we still have PRs missing tests or breaking existing functionality.  Your comment about there being no news is incorrect because we posted [this issue](https://github.com/angular-ui/bootstrap/issues/6404) in early January about contributors/maintainers.  At that point, only @wesleycho was able to donate [a very limited amount of] time.

For us to add anyone as a contributor, we need to see a solid history of his/her contribution work to learn what type of project ambassador he/she would be.  Your comment about having a PR open since March gives the optics, fair or not,  of you just wanting your PR merged so you can move on.  And as I mentioned before, many PRs added simply do not include tests, breaks existing functionality of the library, or both, so them languishing is not an indication of actual contribution.  One needs to understand the full ramifications of a code change before just accepting it and that comes with time.  Time that, unfortunately, no maintainer now has.

**Comment by erikbarke** [comment]
@icfantv, I completely understand (and appreciate...) that you and your team have spent countless hours of your free time nurturing this project, but I also understand that it's time for you guys to move on, that's life.

My problem now is that our team maintains and develops several AngularJS applications with tens of thousands of lines of code and an upgrade (&ast;cough&ast; rewrite &ast;cough&ast;) to Angular 4 or even moving to an another framework simply won't happen, because reasons.

So we're stuck using Angular 1.x and by the looks of it we're not alone; according to builtwith.com there are about half a million live web sites using AngularJS and judging by the amount of stars this project has here on github there are probably tens (hundreds?) of thousands of projects relying on your package. Heck, you even have about half a million of downloads per month on npmjs.com!

This being an open source project means you have no obligations to anyone; you can do whatever you want to, period. All I can do is urge you to don't just abandon the project like this, don't just call it quits and walk away, leaving thousands of developers up the creek without a paddle...

You have already decided that you need to move on and I'm not arguing with that. What rubs me the wrong way is that you're leaving without handing the project over to a new team of maintainers. Sure, anyone could fork the project and keep going but it would be a lot easier if you let someone else take over as a maintainer.

Here's to hoping this project will find a new maintainer... 🍺 
Thanks for all your hard work, we're sad to see you go ☺️

**Comment by zacronos** [comment]
@icfantv I humbly offer some constructive criticism, which I hope does not come across as harsh.

TL;DR: Acknowledge that the current situation isn't ideal, revise the project status notice in the README to include a call for contributors and maintainers, and put a call for contributors and maintainers on https://angular-ui.github.io/bootstrap/ as well.

The optics of the current situation, fair or not, paint the picture that you (the current maintainers) are not terribly interested in handing off this project or seeing bugs get fixed.  Giving you the benefit of the doubt, I assume that's not actually what you want, so let me explain why it looks that way.

On the one hand, you are saying that you want to be very discriminating about who you add as a contributor/maintainer.  This makes sense, and I'm sure is rooted in the desire for this project to continue to succeed.  You don't want quality and community support to fall off because of poor leadership.  I get that, and I get why you want to be careful.

On the other hand, the result is that the project has had a long period without active leadership or progress.  Plus, the community has had no way to gauge how long it will take before the someone else is allowed to take leadership and start making progress again.  That's bad for the project too.

The fact that your responses here don't acknowledge the downside of having no active leadership or progress implies (whether true or not) that you don't understand that downside.  It would help a lot with appearances if you simply said something like "We realize the current situation is not ideal and would very much like to see this project flourish under someone else's leadership.", especially when discussing your standards for adding someone as a contributor.  For instance, it would have provided an entirely different context if you had said that and then "However," before continuing with "for us to add anyone as a contributor...".

I see the call for contributors and maintainers from back in January, but please realize that right now that issue is at the very end of page 5 out of 11 pages of open issues.  Unless someone is actually looking for that or just happens to get lucky with a search term, no one will see it.  Based on the dates on the subsequent open issues, that issue got bumped off page 1 over 8 months ago, so it's been out of easy sight for quite a while.  Even worse, the project status notice at the top of the README says "this project is considered feature-complete and is no longer being maintained".  The way that reads, it sounds like you (the current maintainers) don't think there is even a need for bug or documentation fixes, and strongly implies you are NOT in fact seeking any additional contributors/maintainers.  This creates the appearance, fair or not, that you're either unaware that there is room for bugfixes and other improvements, or that you just don't care whether or not the project gets them.  And that appearance is what leads to calls for project forks, like this one.

I strongly suggest someone revise the project status notice to state clearly that there IS a need for maintenance and bugfixes, and that you are seeking experienced OSS contributors/maintainers.  Maybe even put a link to that January issue so that anyone interested can easily speak up.  Those things should also be added to https://angular-ui.github.io/bootstrap/, since that probably gets some visibility the github page does not.  It would go a long way toward improving appearances, not to mention improving your chances of having someone step up to take over the project.

I am 100% willing to flesh out and PR the README changes I just suggested, if you think it will actually get reviewed and merged.

**Comment by erikbarke** [comment]
@Foxandxss, @pkozlowski-opensource, @wesleycho, you guys seem to be the owners, the main contributors and the publishers of this repo, please help us continue maintaining this repo!

**Comment by elsewares** [comment]
I want to echo the same sentiments as @zacronos - if you're interested in this project not just dying on the vine, then a call for maintainers and contributors needed to happen months ago to get it passed off. This project - ui-bootstrap is *not* feature-complete, and in fact has many outstanding bugs, even taking out of consideration Bootstrap 4 changes, which are legion (and I wouldn't expect to be tackled if you're passing off the maintenance as it is).

If you're not interested in handing off this repo, I suggest both the steps that @zacronos calls for, AND that you call for, and adopt an official fork that you forward people to in the README, instead of just telling current users of the repo, and newcomers, quite incorrectly, that the project is bug-free with Angular 1.x and Bootstrap 3.x.

**Comment by titocr** [comment]
This should be part of a case-study in how not to bail on a popular repo. There have certainly been others, so this isn't to single out this one project. It should be part of a larger general discussion. This is like the dark-side of open-source (probably another dark-side is being a maintainer). Is there a package.json value that 'npm install' or 'npm outdated' could pick up on to give a heads up statuses like 'call-for-maintainers', 'deprecated', 'abandoned', etc.

**Comment by icfantv** [comment]
Just a note that I will delete comments that are simply complaining and venting as they are not constructive.  Options were presented on how to move forward with this project.  Whether or not they are accepted or agreed with is beyond our control.  **If anyone wants to fork this repo and start actively maintaining it, I am more than happy to leave notes, update READMEs, titles, etc... pointing people to that repo.**

For those that complain that calls for maintainers should have been made, you are one of the reasons why maintaining a popular OSS repo is so hard.  Namely, you don't do your due diligence before blasting off a comment, you don't read, and you rely on the project maintainers to do all the investigative legwork and force us to waste our time linking in where those calls were made.  Multiple calls for maintainers were made starting a LONG time ago.

**Comment by stevenvachon** [comment]
Any plans to support AngularJS 1.7?

**Comment by jastend** [comment]
@stevenvachon Angular decided to revert the breaking change in version 1.7.1 with v1.7.2. Just make sure you update to the latest version and you should be all good.

https://github.com/angular/angular.js/issues/16594#issuecomment-396208825

**Comment by arrowd** [comment]
So, any "canonical" fork to contribute to?

**Comment by phazei** [comment]
Here's a continuation of this project that works pretty well.  It has a couple bugs that could probably be handled pretty well if there were some more people supporting it.  It's been updated to use Bootstrap 4:
https://github.com/Morgul/ui-bootstrap4

**Comment by PowerKiKi** [comment]
@icfantv would you agree to mark this repository as [archived](https://help.github.com/en/articles/about-archiving-repositories) ?

There has not been any commits for ~1.5 year, and it would make it even clearer that this project is no longer maintained.

I can do that, if you agree.

**Comment by icfantv** [comment]
@PowerKiKi TBH, I'm not really sure it's necessary as the number of new PRs and Issues filed at this point is virtually non-existent (and the closing of all open issues/tickets feels heavy-handed to me). 
 Also, I don't want to push out a new release that only contains an updated README just to mention the project is being archived since the last time we released, all we did was update the README to say we weren't working on the project anymore.

**Comment by PowerKiKi** [comment]
Publishing a new release isn't required to archive a repo. Actually since the README already contains this kind of info, I wouldn't even touch it at all. So it really just a click on a button to convey without any doubt the status of the project. But that's up to you. I just wanted to make sure that you were aware of the possibility.

**Comment by icfantv** [comment]
Oh, ok.  I'll ping the other two devs to see if they care and if they don't (which they probably won't), we can archive.  Thanks.
